cli/tools/upgrade.rs

```diff
@@ -426,47 +424,21 @@ async fn download_package(
   download_url: &str,
 ) -> Result<Vec<u8>, AnyError> {
   log::info!("Downloading {}", &download_url);
-
-  let res = client.get(download_url).send().await?;
```

Comment:
It seems like too much work/more error prone to maintain this custom code here. Let's just use `http_client.download_with_progress`.

1. No progress output is given when not a tty.
1. No final 100% output is given after downloading, but people can see the sizes while downloading.